tsdb/executor.go

```diff
@@ -625,23 +625,23 @@ func (e *SelectExecutor) processFunctions(results [][]interface{}, columnNames [
 }
 
 func (e *SelectExecutor) processSelectors(results [][]interface{}, callPosition int, hasTimeField bool, columnNames []string) ([][]interface{}, error) {
-	for i, vals := range results {
-		for j := 1; j < len(vals); j++ {
-			switch v := vals[j].(type) {
+	// if the columns doesn't have enough columns, expand it
+	for i, columns := range results {
+		if len(columns) != len(columnNames) {
```

Comment:
That's to be expected, it took me 2 hours to understand the problem.  The real issue is that we don't do enough pre-processing at the query stage to really understand what the shape of the results should be until to late in the process.  This forces us to do some extra work by looping over every call field that could of existed.  

This isn't actually what fixes the problem, but it made more sense to be here than in the following call.  I'll comment below on what the fix is and why it fixes it.